### PR TITLE
Update to core 12.12.0

### PR DESCRIPTION
**Pull request by edualonso**
Also fixed a test that was causing timeouts due to a race condition.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,37 @@
+## 1.5.1 (YYYY-MM-DD)
+
+### Breaking Changes
+* None.
+
+### Enhancements
+* None.
+
+### Fixed
+* None.
```
Comment (cmelchior) [inline CHANGELOG.md:10]: We should probably add the fixes similar to JS: https://github.com/realm/realm-js/blob/master/CHANGELOG.md#fixed-1

File: packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncSessionTests.kt [hunk 2]
```diff
@@ -525,6 +526,11 @@ class SyncSessionTests {
         val realm1 = Realm.open(config1)
         assertNotNull(realm1)
 
+        // Make sure to sync the realm with the server before opening the second instance
+        runBlocking {
+            realm1.syncSession.uploadAllLocalChanges(1.minutes)
```
Comment (cmelchior) [inline packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncSessionTests.kt:531]:
```suggestion
            assertTrue(realm1.syncSession.uploadAllLocalChanges(1.minutes))
```

**Review by cmelchior — APPROVED** [review]
If CI is happy

**Review by elle-j — APPROVED** [review]
Nice, LGTM :)

**Pull request merged**